src/models/tiles/table/table-content.ts

```diff
@@ -450,3 +451,24 @@ export const TableContentModel = TileContentModel
   }));
 
 export type TableContentModelType = Instance<typeof TableContentModel>;
+
+export function updateTableContentWithNewSharedModelIds(
+  content: TableContentModelType,
+  sharedDataSetEntries: PartialSharedModelEntry[],
+  updatedSharedModelMap: Record<string, UpdatedSharedDataSetIds>
+) {
```

Comment:
As I wrote before, I think this is going to cause dependency problems.

In addition I'm surprised that it is taking a `TableContentModelType`. That would represent an instance of the shared model content. But then it is return something that looks like a snapshot with `{...content, columnWidths}`. If it was actually passed an instance then using `...content` to get the properties of that instance would probably not create what we want. I believe that would include any "view getters" the instance has which shouldn't be something included in a snapshot.